src/solvers/dgmulti/sbp.jl

```diff
@@ -237,3 +237,27 @@ function Base.show(io::IO, rd::RefElemData{NDIMS, ElementType, ApproximationType
   @nospecialize rd
   print(io, "RefElemData{", summary(rd.approximationType), ", ", rd.elementType, "}")
 end
+
+function StartUpDG.inverse_trace_constant(rd::RefElemData{NDIMS, ElementType, ApproximationType})  where {NDIMS, ElementType<:Union{Line, Quad, Hex}, ApproximationType<:AbstractDerivativeOperator}
+  D = rd.approximationType
+
+  # this returns the one-dimensional mass matrix, independently of the
+  # dimension of `rd::RefElemData`.
+  M = SummationByPartsOperators.mass_matrix(D)
+  m_diag = diag(M)
+
+  # the inverse trace constant is the maximum eigenvalue corresponding to
+  #       M_f * v = λ * M * v
+  # where M_f is the face mass matrix and M is the volume mass matrix.
+  # Since M is diagonal and since M_f is just the boundary "mask" matrix
+  # (which extracts the first and last entries of a vector), the maximum
+  # eigenvalue is the inverse of the first or last mass matrix diagonal.
+  max_eigenvalue = max(inv(m_diag[1]), inv(m_diag[end]))
```

Comment:
```suggestion
  # the inverse trace constant is the maximum eigenvalue corresponding to
  #       M_f * v = λ * M * v
  # where M_f is the face mass matrix and M is the volume mass matrix.
  # Since M is diagonal and since M_f is just the boundary "mask" matrix
  # (which extracts the first and last entries of a vector), the maximum
  # eigenvalue is the inverse of the first or last mass matrix diagonal.
  left_weight = SummationByPartsOperators.left_boundary_weight(D)
  right_weight = SummationByPartsOperators.right_boundary_weight(D)
  max_eigenvalue = max(inv(left_weight), inv(right_weight))
```
That's faster and doesn't allocate a full 1D mass matrix.

Reply:
That's cool - I didn't know there was that option!